### DISCUSSION TITLE
API level error while uploading bundle to Play Store

### DISCUSSION THREAD

**Discussion by Dylan190774** [quasarframework/quasar#14579] [CLI - Capacitor / Cordova modes]
I build a Quasar Capacitor app with Android Studio.
After uploading the generated bundle to the Play Store for publishing, I get this error:

_Your app currently targets API level 30 and must target at least API level 31 to ensure it is built to the latest API optimized for security and performance. Change your app's target API level to at least 31."_

I'm confused how to fix this. With Cordova this could be done with adding some lines in `config.xml`, but I don't see where to change the `minSDKversion `with Capacitor

**Comment by Dylan190774** [comment]
Upgrading to Capacitor v4 resolves this issue, although not yet officially supported by Quasar.
Update these entries in `package.json` in `src-capacitor` folder:

```
    "@capacitor/cli": "^4.0.0",
    "@capacitor/core": "^4.0.0"
```
If you have added plugins, these need to match their v4 version-number.
Remove `yarn.lock` file and rebuild with `yarn`

**Comment by selfee-jp** [comment]
hello.
I had a problem too.

AndroidManifest.xml
Add "android:exported="true""
```
 <activity
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|smallestScreenSize|screenLayout|uiMode"
            android:name="com.app.****.****.****"
            android:label="@string/title_activity_main"
            android:theme="@style/AppTheme.NoActionBarLaunch"
            android:launchMode="singleTask"
            android:screenOrientation="sensor"
            android:exported="true"
            >

            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>

        </activity>
```
Gradle Scripts > variables.gradle
Rewrite "compileSdkVersion" and "targetSdkVersion" to 31.
```
    minSdkVersion = 21
    compileSdkVersion = 31
    targetSdkVersion = 31
```

Build it and upload it.

**Reply by selfee-jp** [reply]
However, push notifications do not work in the background. 
How can this be resolved?
